zerver/webhooks/newrelic/doc.md

```diff
@@ -1,27 +1,81 @@
-New Relic can send messages to a Zulip channel for incidents.
+# Zulip New Relic integration
+
+Get Zulip notification for New Relic incidents.
+
+{start_tabs}
 
 1. {!create-channel.md!}
 
 1. {!create-an-incoming-webhook.md!}
 
-1. {!generate-integration-url.md!}
+1. {!generate-webhook-url-basic.md!}
+
+1. In New Relic, go to the **Alerts** menu, and select **Destinations**.
+   Choose **Webhook** in the **Add a destination** section.
 
-1. On [New Relic](https://one.newrelic.com),
-  select **Alerts & AI**.
+1. Set a **Webhook name**, such as `Zulip`. Set the **Endpoint URL** to
+   the URL generated above. Click **Save destination**.
 
-1. Navigate to **Notification channels**.
+1. In the **Alerts** menu, select **Workflows**. Click on
+   **+ Add a Workflow**.
 
-1. Create a new notification channel. Select channel type of **Webhook**, choose a name (e.g., "Zulip"), enter the webhook url created earlier as **Base Url**.
+1. Set your workflow name, and filter the trigger conditions. In the
+   **Notify** section, choose **Webhook**. In the **Edit notification
+   message** menu, select the destination for Zulip created above.
 
-1. It should look like:
-  ![](/static/images/integrations/newrelic/newrelic.png)
+1. In the **Payload** section, you can configure the payload for this
+   workflow. The default payload template is sufficient to get the
+   integration working, but using the message template below will enable
+   the integration to notify you of any **acknowledged** New Relic
+   incidents. To include additional custom fields, refer to
+   [configuration options](#configuration-options):
 
-1. The webhook works with the default payload, click **Create channel**.
+         {
+            {% raw %}
+            "id": {{ json issueId }},
+            "issueUrl": {{ json issuePageUrl }},
+            "title": {{ json annotations.title.[0] }},
+            "priority": {{ json priority }},
+            "totalIncidents": {{json totalIncidents}},
+            "state": {{ json state }},
+            "createdAt": {{ createdAt }},
+            "closedAt": {{ json closedAt }},
+            "updatedAt": {{ updatedAt }},
+            "alertPolicyNames": {{ json accumulations.policyName }},
+            "alertConditionNames": {{ json accumulations.conditionName }},
+            "acknowledgedAt": {{ json acknowledgedAt }},
+            "acknowedgedBy":{{ json acknowledgedBy }},
```

Comment:
Just noticed that this is misspelled (acknowedged vs acknowledged) both here in the documentation and in the example fixtures.

Also, I don't think this field is used currently in the view code for building the message. If we're not using it, then it should probably be removed from the documentation. If we want to use it (maybe like we use the owner field), then we should update the code to do so. I don't think it's in the expected fields group.

Reply:
Ah, this is another field that should be left out. It only provides the users ID

Reply:
Removed a couple more fields that we don't use from the provided base template, here's an example from one file:

<details>

<summary>Git Diff</summary>

```diff
-- a/zerver/webhooks/newrelic/fixtures/incident_with_invalid_zulip_custom_fields.j
son
+++ b/zerver/webhooks/newrelic/fixtures/incident_with_invalid_zulip_custom_fields.j
son
@@ -3,28 +3,19 @@
     "issueUrl": "https://radar-api.service.newrelic.com/accounts/4420147/issues/13bbc
dca-f0b6-470d-b0be-b34583c58869?notifier=WEBHOOK",
     "title": "PIETER-UBUNTU query result is > 1.0 for 5 minutes on 'Storage on Host E
xceeded Threshold'",
     "priority": "HIGH",
-    "impactedEntities": [
-      "PIETER-UBUNTU"
-    ],
     "totalIncidents": 1,
     "state": "ACTIVATED",
     "trigger": "STATE_CHANGE",
-    "isCorrelated": "false",
     "createdAt": 1713769949493,
     "closedAt": "N/A",
     "updatedAt": 1713769949494,
-    "sources": [
-      "newrelic"
-    ],
     "alertPolicyNames": [
       "Golden Signals"
     ],
     "alertConditionNames": [
       "Storage on Host Exceeded Threshold"
     ],
-    "workflowName": "issue workflow",
     "acknowledgedAt": "N/A",
-    "acknowedgedBy": "N/A",
     "owner": "N/A",
     "zulipCustomFields": {
       "Invalid fields 1": ["SSD", 2000, false, null, {}],

```

</details>